src/main/java/com/cloudbees/jenkins/plugins/customtools/CustomToolInstallWrapper.java

```diff
@@ -102,174 +91,159 @@ public String toString() {
     }
     
     private @Nonnull SelectedTool[] selectedTools = new SelectedTool[0];
-    private final @CheckForNull MulticonfigWrapperOptions multiconfigOptions;    
-    private final boolean convertHomesToUppercase;
-    
+    private @CheckForNull MulticonfigWrapperOptions multiconfigOptions = MulticonfigWrapperOptions.DEFAULT;
+    private boolean convertHomesToUppercase = false;
+    private transient UsedCustomToolAction usedCustomToolAction;
+
     @DataBoundConstructor
-    public CustomToolInstallWrapper(SelectedTool[] selectedTools, MulticonfigWrapperOptions multiconfigOptions, boolean convertHomesToUppercase) {
+    public CustomToolInstallWrapper(SelectedTool[] selectedTools) {
         this.selectedTools = (selectedTools != null) ? selectedTools : new SelectedTool[0];
+    }
+
+    @DataBoundSetter
+    public void setMulticonfigOptions (MulticonfigWrapperOptions multiconfigOptions){
         this.multiconfigOptions = (multiconfigOptions != null) ? multiconfigOptions : MulticonfigWrapperOptions.DEFAULT;
+    }
+
+    @DataBoundSetter
+    public void setConvertHomesToUppercase (boolean convertHomesToUppercase) {
         this.convertHomesToUppercase = convertHomesToUppercase;
     }
     
     public boolean isConvertHomesToUppercase() {
         return convertHomesToUppercase;
-    }   
-    
-    @Override
-    public Environment setUp(AbstractBuild build, Launcher launcher,
```

Comment:
There is some large refactoring that needs to be done here, but to take care of some of the pipeline unrelated diff consider #28